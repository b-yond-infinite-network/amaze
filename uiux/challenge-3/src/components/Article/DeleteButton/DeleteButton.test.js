import DeleteButton from './DeleteButton';
import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer'
import configureMockStore from 'redux-mock-store'
import middlewares from '../../../store'

describe('DeleteButton Component', () => {

    let mockStore;

    let tree;
    let wrapper;
    let render;

    describe('show === true', () => {

        beforeAll(() => {

            mockStore = configureMockStore(middlewares)({});

            const props = {
                show: true
            };

            tree = <DeleteButton store={mockStore} {...props} />;
            wrapper = mount(tree);
            console.log(wrapper.debug());
            render = renderer.create(tree).toJSON();

        });

        it('Should match the snapshot', () => {
            expect(render).toMatchSnapshot();
        });

        it('Should show the delete button', () => {
            const component = wrapper.find('#delete-button');
            expect(component.length).toBe(1);
        });

    });

    describe('show === false', () => {

        beforeAll(() => {

            mockStore = configureMockStore(middlewares)({});

            const props = {
                show: false
            };

            tree = <DeleteButton store={mockStore} {...props} />;
            wrapper = mount(tree);
            console.log(wrapper.debug());
            render = renderer.create(tree).toJSON();

        });

        it('Should match the snapshot', () => {
            expect(render).toMatchSnapshot();
        });

        it('Should NOT show the delete button', () => {
            const component = wrapper.find('#delete-button');
            expect(component.length).toBe(0);
        });

    });

});