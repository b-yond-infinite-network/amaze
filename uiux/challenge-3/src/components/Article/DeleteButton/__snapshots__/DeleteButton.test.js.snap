// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteButton Component show === false Should match the snapshot 1`] = `null`;

exports[`DeleteButton Component show === true Should match the snapshot 1`] = `
<span
  className="mod-options"
  id="delete-button"
>
  <i
    className="ion-trash-a"
    onClick={[Function]}
  />
</span>
`;
